src/main/java/com/forgeessentials/api/UserIdent.java

```diff
@@ -214,39 +271,111 @@ public static synchronized UserIdent get(EntityPlayerMP player)
         else
         {
             String name = player.getCommandSenderName();
-            if (name != null && !name.equals(ident.username))
+
+            String newname = name + "_" + ident.uuid.hashCode();
+            if (name != null && !name.equals(ident.username) && !newname.equals(ident.username))
             {
-                byUsername.remove(ident.username);
+                if (byUsername.containsKey(name.toLowerCase()))
```

Comment:
The issue is that fake players can also change their name.  And if the name already exists you get a duplicate player.